internal/cmd/test/test.go

```diff
@@ -1,232 +1,46 @@
-			fm, err := pop.NewFileMigrator("./migrations", test)
-			if err != nil {
+var (
+	//go:embed longhelp.txt
+	longHelp string
+	Cmd      = &cobra.Command{
+		Use:   "test",
+		Short: "Runs tests for your Buffalo app",
+		Long:  longHelp,
+
+		// DisableFlagParsing is set to true since we will need to allow undefined
+		// flags to be passed to the go test command.
+		DisableFlagParsing: true,
+
+		RunE: func(c *cobra.Command, args []string) error {
+			// Set the environment to be test so that the rest of the tooling
+			// understands we're in testing mode.
+			if err := os.Setenv("GO_ENV", "test"); err != nil {
 				return err
 			}
 
-			if err := fm.Up(); err != nil {
+			// Setup the database before running the tests.
+			if err := setupDatabase(args); err != nil {
 				return err
 			}
 
-			return testRunner(args)
-		}
-
-		if schema := findSchema(); schema != nil {
-			err = test.Dialect.LoadSchema(schema)
+			tcmd, err := buildCmd(args)
 			if err != nil {
 				return err
 			}
-		}
-	}
-	return testRunner(args)
-}
-
-func findSchema() io.Reader {
-	if f, err := os.Open(filepath.Join("migrations", "schema.sql")); err == nil {
-		return f
-	}
-	if dev, err := pop.Connect("development"); err == nil {
-		schema := &bytes.Buffer{}
-		if err = dev.Dialect.DumpSchema(schema); err == nil {
-			return schema
-		}
-	}
-
-	if test, err := pop.Connect("test"); err == nil {
-		fm, err := pop.NewFileMigrator("./migrations", test)
-		if err != nil {
-			return nil
-		}
-
-		if err := fm.Up(); err == nil {
-			return nil
-		}
-	}
-	return nil
-}
-
-func testRunner(args []string) error {
-	var mFlag bool
-	var query string
-
-	commandArgs := []string{}
-	packageArgs := []string{}
-
-	var lastArg string
-	for index, arg := range args {
-		switch arg {
-		case "-run", "-m":
-			query = args[index+1]
-			mFlag = true
-		case "-v", "-timeout":
-			commandArgs = append(commandArgs, arg)
-		default:
-			if lastArg == "-timeout" {
-				commandArgs = append(commandArgs, arg)
-			} else if lastArg != "-run" && lastArg != "-m" {
-				packageArgs = append(packageArgs, arg)
-			}
-		}
-
-		lastArg = arg
-	}
-
-	cmd := newTestCmd(commandArgs)
-	if mFlag {
-		return mFlagRunner{
-			query: query,
-			args:  commandArgs,
-			pargs: packageArgs,
-		}.Run()
-	}
-
-	pkgs, err := testPackages(packageArgs)
-	if err != nil {
-		return err
-	}
 
-	cmd.Args = append(cmd.Args, pkgs...)
-	logrus.Info(strings.Join(cmd.Args, " "))
-	return cmd.Run()
-}
-
-type mFlagRunner struct {
-	query string
-	args  []string
-	pargs []string
-}
-
-func (m mFlagRunner) Run() error {
-	app := meta.New(".")
-	pwd, _ := os.Getwd()
-	defer os.Chdir(pwd)
-
-	pkgs, err := testPackages(m.pargs)
-	if err != nil {
-		return err
-	}
-
-	var errs bool
-	for _, p := range pkgs {
-		os.Chdir(pwd)
-
-		if p == app.PackagePkg {
-			continue
-		}
-
-		cmd := newTestCmd(m.args)
-
-		p = strings.TrimPrefix(p, app.PackagePkg+string(filepath.Separator))
-		os.Chdir(p)
-
-		if hasTestify(cmd.Args) {
```

Comment:
We previously used this "for each package loop" approach as a workaround for the `-testify.m` conditions:
- just with the `-testify.m TestMethod` only without `-run TestSuite`,
- per package switching of the flag between `-run` and `-testify.m` with the (quick and dirty) `hasTestify()` helper.

Reply:
Right. IMO we should move to run tests closer to the standard library. At this point, just to support this -testify.m behavior we're disallowing users from passing arguments to the `go test` command.